bundles/org.openhab.binding.opensmartcity/src/main/java/org/openhab/binding/opensmartcity/internal/OpenSmartCityWeatherHandler.java

```diff
@@ -74,67 +90,172 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+                }
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Dispose handler '{}'.", getThing().getUID());
         ScheduledFuture<?> localRefreshJob = refreshJob;
-        if (localRefreshJob == null || localRefreshJob.isCancelled()) {
-            logger.debug("Start refresh job at interval {} seconds.", bridgeHandler.config.refreshInterval);
-            refreshJob = scheduler.scheduleWithFixedDelay(this::updateSensorValues, 0,
-                    bridgeHandler.config.refreshInterval, TimeUnit.SECONDS);
+        if (localRefreshJob != null && !localRefreshJob.isCancelled()) {
+            logger.debug("Stop refresh job.");
+            if (localRefreshJob.cancel(true)) {
+                refreshJob = null;
+            }
         }
     }
 
-    private void updateSensorValues() {
+    public class JsonResponse {
+    }
+
+    private String apiRequest(String queryPath) {
         try {
-            String url = bridgeHandler.basePath + QUERY_PATH;
+            String url = bridgeHandler.basePath + queryPath;
             logger.debug("Requesting {}", url);
 
             Request request = bridgeHandler.httpClient.newRequest(url);
-
             // TODO: remove hard coded credentials
             request.getHeaders().add("Authorization",
                     "Basic " + Base64.getEncoder().encodeToString(("smarthomeuser:Solingen2030!").getBytes()));
             ContentResponse response = request.send();
 
             if (response.getStatus() == 200) {
-                updateStatus(ThingStatus.ONLINE);
-
                 String content = response.getContentAsString();
-                JsonObject jsonResponse = gson.fromJson(content, JsonObject.class);
-                logger.debug("Response: {}", jsonResponse.toString());
-                // JsonElement observations = jsonResponse.get("value").getAsJsonArray().get(0);
-                Double temperature = 11.3;
-                Double humidity = 0.85;
-
-                updateState(OpenSmartCityBindingConstants.CHANNEL_TEMPERATURE,
-                        new QuantityType<Temperature>(temperature, SIUnits.CELSIUS));
-                updateState(OpenSmartCityBindingConstants.CHANNEL_HUMIDITY, new DecimalType(humidity));
-
-            } else {
-                // TODO: check exact problem
-                updateStatus(ThingStatus.OFFLINE);
-                logger.debug("HTTP request failed with response code {}: {}", response.getStatus(),
-                        response.getReason());
+                logger.debug("Requesting URL {}", url);
+                logger.debug("Retreived Content {}", content);
+                return content;
             }
         } catch (InterruptedException | ExecutionException | TimeoutException e) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
         }
+        return "";
```

Comment:
Wouldn't it better to return `null` here?

Reply:
Changed